include/cuco/static_map.cuh

```diff
@@ -945,10 +941,36 @@ class static_map {
      *
      * Returns a pair consisting of an iterator to the inserted element (or to
      * the element that prevented the insertion) and a `bool` denoting whether
-     * the insertion took place. Uses the CUDA Cooperative Groups API to
-     * to leverage multiple threads to perform a single insert. This provides a
-     * significant boost in throughput compared to the non Cooperative Group
-     * `insert` at moderate to high load factors.
+     * the insertion took place.
+     *
+     * Note: This function is slower than `insert` for non-packable key/value
+     * types because it has to ensure a valid iterator in case of duplicates.
+     * For the same reason, the function must not be called concurrently with
+     * `insert` unless `supports_concurrent_insert_find()` returns true.
```

Comment:
"packable" types is an implementation detail. It's sufficient to just say that the performance may be worse and to prefer the other flavor of `insert` when possible. 


```suggestion
     * Note: In order to guarantee the validity of the returned iterator, `fetch_insert` may be less efficient than `insert` in some situations. Prefer using `insert` unless the returned iterator is required.
     *
     * Note: `fetch_insert` may only be used concurrently with `insert`, `find`, and `erase` when `supports_concurrent_insert_find()` returns true. 
```